cmd/main.go

```diff
@@ -145,19 +149,25 @@ func runScan(ctx context.Context, c config.Config, client scan.EthClient, s scan
 				slog.Error("error retrieving the latest block", "err", err.Error())
 				break
 			}
-			lastBlock := calculateLastBlock(c.StartingBlock, l1LatestBlock, c.BlocksRange, c.BlocksMargin)
-			if lastBlock < c.StartingBlock {
+			lastBlock := calculateLastBlock(startingBlock, l1LatestBlock, c.BlocksRange, c.BlocksMargin)
+			if lastBlock < startingBlock {
```

Comment:
This logic stems from a previous PR, wherein we encountered the same issue and made a deliberate choice for the following reason: If we were to relocate this logic from its current function, we would lose the ability to include the 'slog.Debug' component. While it's possible to handle logging in a different function, the ultimate function responsible for recording any form of information should remain the same.

Reply:
To be honest, I would place this check inside the function. Also, I don't understand why you cannot place the log inside the function. But I won't push this if you want to merge PR and finish your task

Reply:
We decided to create a function to calculate the last block as it was too verbose to show it in the caller code.
We decided not to do the same with `lastBlock < startingBlock` as it's just a simple comparison and it's easy to understand for a reader.

We could move the comparison within `calculateLastBlock`, but then the method would do three things: calculate the last block, compare it with the starting block, and return it. Finding a name for that function won't be easy, and, in my personal opinion, it would be counterintuitive for a reader to look at a method that both calculates and compares, while also returning a piece of data.

Regarding the `slog.Debug` call: we think it's the responsibility of the caller to do that as it knows the flow of execution, the underlying function doesn't have to know that when the last block is behind the starting block, the caller wants to wait.